### DISCUSSION TITLE
how to disable asm language for openssl in ics-openvpn android build

### DISCUSSION THREAD

**Discussion by FxamarAboali** [openssl/openssl#23222] [Q&A]
hi
l  am edited some classes c .
when I want to build ics-openvpn app, I can't see my modify because OpenSSL use asm asmply language by default.
how to disable no-asm language for OpenSSL to build ics-openvpn.

**Comment by slontis** [comment]
See https://github.com/openssl/openssl/blob/master/INSTALL.md#no-asm

**Reply by FxamarAboali** [reply]
Ok , but when I build ics-openvpn android I can't disable asm I don't know who, please help me.

**Reply by t8m** [reply]
I am sorry but we do not know what ics-openvpn android is and how to change OpenSSL build options under its build.

**Comment by nhorman** [comment]
I assume ics-openvpn imports the openssl source as part of the build?  If you are asking how to modify the ics-openvpn gradle properties to add no-asm to its build, thats going to be a question for the ics-openvpn project maintainers

**Reply by FxamarAboali** [reply]
yes asking how to modify the ics-openvpn gradle properties to add no-asm to its build.

i asked him but he don't help me free.

please anyone help me

**Comment by slontis** [comment]
I am afraid this is outside the scope of what OpenSSL will answer.
The external answer you were given is that it is complex to change it, as they have written custom .cmake files
All I can suggest you look at the difference between makefiles that files get produced if you try to do a ./Configure no-asm ans ./Çonfigure
It looks like someone has recently added an exporter for CMake  for Unix/Windows (but I dont know much about it).
